src/main/java/seedu/cafectrl/storage/Storage.java

```diff
@@ -23,19 +23,61 @@ public Storage (Ui ui) {
         this.ui = ui;
     }
 
+    //@@author Cazh1
+    /**
+     * Boolean to detect if the text save file has been tampered with
+     *
+     * @param encodedStringArrayList The arraylist of string read from text save file
+     * @return true is the file's hash is not normal or does not match the newly generated hash, false otherwise
+     */
+    private boolean isFileCorrupted(ArrayList<String> encodedStringArrayList) {
+        //Hash string is stored as last in the ArrayList
+        int lastIndex = encodedStringArrayList.size() - 1;
+        String hashString = encodedStringArrayList.get(lastIndex);
+
+        //Checks if the saved Hash is abnormal
+        if (((!hashString.matches("^[0-9]+$")) && (!hashString.matches("^-[0-9]+$"))) ||
+                hashString.matches("^0{2,}$")) {
+            //logger.log(Level.INFO, "Corrupted data file");
+            return true;
+        }
+
+        int fileHash = Integer.parseInt(hashString);
+        //Removes the saved Hash String for decoding
+        encodedStringArrayList.remove(lastIndex);
+
+        //Prepares String in same format as when encoding, generates Hash from the save file content
+        String encodedMenuAsString = String.join(", ", encodedStringArrayList).trim();
+        int encodedMenuHash = encodedMenuAsString.hashCode();
+
+        //Checks if the generated Hash matches the saved Hash
+        if (encodedMenuHash != fileHash) {
+            return true;
+        }
+        return false;
+    }
+
     //@@author ShaniceTang
     /**
      * Loads menu data from a text file, decodes it, and returns it as a Menu object.
      *
      * @return A Menu object containing data from the file.
      */
-    public Menu loadMenu() {
+    public Menu loadMenu(){
```

Comment:
Maybe add a whitespace here too? :)

```suggestion
    public Menu loadMenu() {
```